sway-fmt-v2/src/utils/comments.rs

```diff
@@ -0,0 +1,182 @@
+use crate::fmt::{Formatter, FormatterError};
+use std::{cmp::Ordering, collections::BTreeMap, fmt::Write, sync::Arc};
+use sway_parse::token::{lex_commented, Comment, CommentedTokenTree, CommentedTree};
+
+/// Represents a span for the comments in a spesific file
+/// A stripped down version of sway-types::src::Span
+#[derive(PartialEq, Eq, Debug, Clone, Default)]
+pub struct CommentSpan {
+    // The byte position in the string of the start of the span.
+    pub start: usize,
+    // The byte position in the string of the end of the span.
+    pub end: usize,
+}
+
+impl Ord for CommentSpan {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // If the starting position is the same encapsulatig span (i.e, wider one) should come
+        // first
+        match self.start.cmp(&other.start) {
+            Ordering::Equal => self.end.cmp(&other.end),
+            ord => ord,
+        }
+    }
+}
+
+impl PartialOrd for CommentSpan {
+    fn partial_cmp(&self, other: &CommentSpan) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+pub type CommentMap = BTreeMap<CommentSpan, Comment>;
+
+/// Get the CommentedTokenStream and collect the spans -> Comment mapping for the input source
+/// code.
+pub fn construct_comment_map(input: Arc<str>) -> Result<CommentMap, FormatterError> {
```

Comment:
I think just `comment_map` should be pretty unambiguous here.

Otherwise, if you do prefer to clarify that it is a constructor, the general convention for constructors in Rust is to use `new` or `from`, e.g. `new_comment_map` or `comment_map_from_src`.